addon_service/authorized_storage_account/models.py

```diff
@@ -68,12 +72,21 @@ class AuthorizedStorageAccount(AddonsServiceBaseModel):
         blank=True,
         related_name="authorized_storage_account",
     )
+    _temporary_oauth1_credentials = models.OneToOneField(
+        "addon_service.ExternalCredentials",
```

Comment:
instead of using `ExternalCredentials` for the temporary token/secret, how about move the encryption logic to a reusable abstract model and add a new table for oauth1 use?

```py
# addon_service/common/encrypted_json_model.py?
class EncryptedJsonModel(AddonsServiceBaseModel):
    class Meta:
        abstract = True

    encrypted_json = models.BinaryField()
    _salt = models.BinaryField()
    ... # encryption-related fields and methods from ExternalCredentials


# addon_service/credentials/models.py
class ExternalCredentials(EncryptedJsonModel):
    # credentials-specific methods only
    ...


# addon_service/oauth1/models.py
class Oauth1TemporaryToken(EncryptedJsonModel):
    # oauth1-specific methods/fields only
    ...
```

Reply:
Can you elaborate on which particular logic you want to have encapsulated in these classes?

Reply:
yep, elaborated the idea (tho not yet to working state) in branch `oauth1-elaboration` -- see `EncryptedDataclassModel`: https://github.com/opaduchak/gravyvalet/pull/1/files#diff-027727e35af18734580549b67475e3bd1d00ba714d205291c254e34eabf32abc

Reply:
i'm fine letting this idea go -- the current implementation does well enough keeping temporary credentials logically separate, even while sharing the ExternalCredentials table, and since [zotero does oauth1 (maybe) slightly wrong](https://github.com/opaduchak/gravyvalet/pull/1#discussion_r1674513336) we can't do the lookup-by-temporary-token that was part of the motivation for an oauth1-specific table